### ISSUE TITLE
Luxor balances missing - Update: Luxor enabled but not listed in "Host" column of "All Pools" page

### ISSUE THREAD

**Issue by smyers63** [RainbowMiner/RainbowMiner#2272]
The balances appeared when I first started using RBM a couple of months ago but stopped not too long afterward.

**Comment by smyers63** [comment]
I've been looking into why the balances are missing.  I wish I was more familiar with writing these scripts or I might see what's wrong.  At any rate, this info might lead to a fix:

https://luxor.zendesk.com/hc/en-us/articles/360019567952-Monitoring-API-Documentation

**Comment by smyers63** [comment]
While looking into the issue, I've also found that even though Luxor is enabled in Setup, it is not listed as a Host on the All Pools page.  Its pool module is still shown as loaded during startup.

![image](https://user-images.githubusercontent.com/95280323/214987203-6b698db3-3ff7-4c37-b8fe-0dfa6cc3f5b5.png)

**Comment by smyers63** [comment]
I reported the problem with Luxor balances many weeks ago and there hasn't been a response to that.  Why is this so?

**Comment by RainbowMiner** [comment]
Ok, now I remember: Luxor pool just hasn't any coins to mine with GPU, currently. RainbowMiner's modules do in fact contain only one entry, it's ARRR. So you can only mine ARRR and see balance of ARRR.

**Comment by smyers63** [comment]
I had added the other mineable coins and have balances on ARRR and 2 others.  None show in RBM, not even ARRR.

![Luxor-Screenshot 2023-01-31 153223](https://user-images.githubusercontent.com/95280323/215888298-5ddd18d8-5073-4279-b3bc-bc3e322c292e.png)

**Comment by smyers63** [comment]
Have you double-checked what you've said about Luxor?  There are other GPU-mineable coins (see screenshot from 2 weeks ago) and I mined 3 of them before RBM stopped using Luxor.

**Comment by RainbowMiner** [comment]
In fact, I did check something, but then put the Luxor on the sidelines. I'll reactivate all coins soon.

**Comment by RainbowMiner** [comment]
Ok, I have been able to invoke the new Luxor API for the mining pool module (commit 8140b60357819f2ee258f3aa1cfb6417e3736eea, you will now have to add an API key to pools.config.txt - the parameter field will appear, after the next release update). Unfortunately there is no longer access to the legacy API and there is no way to read the balance via the new API currently. So I have removed the balance module for this pool until the balance returns to the new API.

**Comment by smyers63** [comment]
Thanks, I'll be looking forward to that.

Looking at the docs, it looks like "revenue" would return the current balance for the coin queried.  That doesn't work?
https://docs.luxor.tech/docs/schema/objects/coin-mining-overview

**Comment by RainbowMiner** [comment]
Oh! I didn't recognize this field as balance. If it's the balance, then that would do it. Let's try.

**Comment by smyers63** [comment]
I still don't see any Luxor balances, Luxor is still missing from "All Pools" and I haven't had any mining there in months.  Have you made any changes yet toward a fix?  I'm about to look for another way to mine there so I can get out the coins I've already mined.

**Comment by smyers63** [comment]
I've added my API key and shut off all pools but Luxor to see if RBM would even mine there again.  Still no mining.  Is the API key related at all to mining or is it just the means to read the stats?

Certainly, I'd like to be able to see the balances in RBM but if I can't mine, the balances don't matter too much.

**Comment by RainbowMiner** [comment]
I am already in talks with the pool developers. They haven't a function to get the current balance in their API yet, but promised to update the software quickly. So the balance should reappear. For which coins did you setup wallets in the pools.config.txt / Luxor section?

**Comment by RainbowMiner** [comment]
> I've added my API key and shut off all pools but Luxor to see if RBM would even mine there again. Still no mining

I am currently reworking that pool module. Update is coming soon.

**Comment by RainbowMiner** [comment]
I just noticed, that we don't have an Equihash GPU miner. Since it's ASIC territory, all Equihash (not the variants, but the ZEN, ARRR coin variant). I can re-add DSTM - that should finally enable your rigs to miner Equihash on Luxor again.

**Comment by smyers63** [comment]
What happened since I mined these 3 with RBM using only GPUs?  They're small amounts so if I had to abandon them it wouldn't hurt much but since it worked then, I'm just wondering what has changed.  I might be able to do something different to mine those again.  My position is that I don't want to give up on a pool/miner/algo combination that might be profitable later even if it isn't right now.

![ZcashScreenshot 2023-03-12 at 20-20-04 Luxor Mining](https://user-images.githubusercontent.com/95280323/224587880-0aa321af-cd98-46d7-9a4a-869e2865652d.png)

![ZENScreenshot 2023-03-12 at 20-20-37 Luxor Mining](https://user-images.githubusercontent.com/95280323/224587894-a1795ae4-b248-4f25-9119-2f855e3e8ce0.png)

![ARRRScreenshot 2023-03-12 at 20-18-43 Luxor Mining](https://user-images.githubusercontent.com/95280323/224588031-dca647a2-d8a8-40af-a240-757ba21f45d1.png)

**Comment by smyers63** [comment]
> I am already in talks with the pool developers. They haven't a function to get the current balance in their API yet, but promised to update the software quickly. So the balance should reappear. For which coins did you setup wallets in the pools.config.txt / Luxor section?

ARRR  (mined before with RBM)
SC 
DASH 
BTC 
ZEC  (mined before with RBM)
ZEN  (mined before with RBM)
ETH 

I know some of those will never mine but I usually add all the coins that I have a wallet for at each pool I decide to use just in case some might be mineable later.  I wouldn't want to find out too late that I'd missed out on something profitable.

**Comment by smyers63** [comment]
I'm doing some research on mining those coins.  Here's a little I've found so far.

[quote]
There are a lot of miner-programs that you can use for the equihash algorithm, which the Pirate blockchain is built on:

    DSTM
    EWBF
    BMiner
    Lolminer for AMD

(to name just a few)
[end quote]

There isn't a date on the article that came from but that might help.

**Comment by smyers63** [comment]
I just tested Equihash at Nicehash, Zergpool and Luxor.  At Luxor I mined on ARRR, ZEC and ZEN.  EWBF CUDA Zcash Miner 0.3.4b worked.  CcMiner 2.3.1 worked but seemed to get stuck pretty quickly.  DSTM didn't work (as I expected from your results).

Even though I've said this already, I want to repeat it in case anyone gets the wrong idea.  I'm not saying these algos/coins/pools are profitable right now with GPUs.  Here's my situation.  There are cases with certain pool/algo/coin combinations where I had mined maybe 50% or more of the payout threshold before I found more profitable coins and pools.  Of course, I started mining the more profitable stuff then and have money left at the previous pools.  I want to get to a payout either now or at some time later when/if that combination is more profitable again.  I don't want to leave money behind but I also don't want to spend more than it's worth to get it.  I just want to be _able_ to get it.

Part of the reason I have balances of some of these low-profit coins is that I was looking for something to replace the income I lost when Ethereum went POS.  I didn't find much, if any, profit until after I started using RBM and ran into some pools and coins I didn't know about before.  The small balances at a few pools aren't much on their own but added up make several dollars I'd like to have.  In cold weather, I don't mind using the GPUs on those now and then.  The cold weather is almost over here for another 9 or 10 months so that's why I've been asking more about it.
